### PR TITLE
Add chunk & manifest storage

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

File: oxidation/crates/parsec_client_types/src/local_manifest.rs [hunk 1]
```diff
@@ -1013,3 +1013,54 @@ impl LocalUserManifest {
...
+    pub fn dump_and_encrypt(&self, key: &SecretKey) -> Vec<u8> {
+        match self {
+            Self::File(manifest) => manifest.dump_and_encrypt(key),
+            Self::Folder(manifest) => manifest.dump_and_encrypt(key),
+            Self::Workspace(manifest) => manifest.dump_and_encrypt(key),
+            Self::User(manifest) => manifest.dump_and_encrypt(key),
+        }
+    }
+
+    pub fn decrypt_and_load(encrypted: &[u8], key: &SecretKey) -> Result<Self, &'static str> {
```
Comment (TimeEngineer) [inline oxidation/crates/parsec_client_types/src/local_manifest.rs:1053]: I will improve this

File: oxidation/libparsec/crates/core_fs/src/error.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use thiserror::Error;
+use uuid::Uuid;
+
+use parsec_api_crypto::CryptoError;
+use parsec_api_types::FileDescriptor;
+
+#[derive(Error, Debug)]
+pub enum FSError {
```
Comment (FirelightFlagboy) [inline oxidation/libparsec/crates/core_fs/src/error.rs:10]: Why the pascal case for the error message ?

Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/error.rs:10]: Is there any reason for another case ?

Comment (FirelightFlagboy) [reply oxidation/libparsec/crates/core_fs/src/error.rs:10]: I prefer phrases instead of word like:

```diff
- ConnectionError
+ connection error
```

Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/error.rs:10]: I like this type of error: `Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }`
Where the PascalCase is the kind of error

Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/error.rs:10]: You can retrieve fastly your error in that case

Comment (TimeEngineer) [reply oxidation/libparsec/crates/core_fs/src/error.rs:10]: #2438 I opened an issue to format Error message

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**